src/sources/datadog_agent/metrics.rs

```diff
@@ -378,10 +378,7 @@ fn decode_datadog_series_v1(
 
 fn into_metric_tags(tags: Vec<String>) -> MetricTags {
     tags.iter()
-        .map(|tag| {
-            let kv = tag.split_once(':').unwrap_or((tag, ""));
-            (kv.0.trim().to_string(), kv.1.trim().to_string())
-        })
+        .map(|tag_chunk| extract_tag_key_and_value(tag_chunk))
```

Comment:
Can you just `.map(extract_tag_key_and_value)` here? If not, could the signature of that function be changed to allow it?